### PR TITLE
Build pipeline using magefile

### PR DESCRIPTION
**Pull request by a-palchikov**
## Description

This PR is based on Kevin's work on magefile-based build pipeline. It also incorporates changes to enable building on macos (10.15.x).
The changes are not supposed to interfere with the existing build pipeline but the plan is to replace it eventually.
The hope is that it's going to be helpful upstream but let me know if that's not the case.  

## Type of change

* New feature (non-breaking change which adds functionality)
* Internal change (not necessarily a bug fix or a new feature)

## Linked tickets and other PRs

* Requires https://github.com/gravitational/planet/pull/838, https://github.com/gravitational/magnet/pull/2, https://github.com/knisbet/mage/pull/2

## TODOs

- [x] Self-review the change
- [ ] Write documentation
- [ ] Address review feedback
- [ ] Update upstream references / tags / versions after upstream PR merges (linked above)

### CODE REVIEW

**Review by knisbet — APPROVED** [review]

File: mage/misc.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+// Env outputs the list of imported environment variables
+func Env() (err error) {
+	for k, v := range root.Config.ImportEnv {
+		fmt.Println(k, ":", v)
+	}
+
+	return
+}
+
+// IsUpToDate returns true iff all of the (source, sources...) are older or the same
```
[the remark this answers is not in the document]
Comment (a-palchikov) [reply mage/misc.go:54]: `iff` is meant as `if and only if` - isn't there such a written abbreviation? Or do you mean it's still inappropriate?

**Comment by a-palchikov** [comment]
@knisbet  @wadells PTAL - is there anything else you'd like to change?

**Comment by wadells** [comment]
> @knisbet @wadells PTAL - is there anything else you'd like to change?

Sorry, I haven't looked at the mage stuff at all since last year -- and this just showed up on my review radar now.  Given the magnitude of the changes, I probably won't be able to provide meaningful feedback until the week of June 7th.  Given the parallel implementation and Kevin's in depth knowledge, I trust his judgement if you want to move forward before then.

File: mage/misc.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (knisbet) [reply mage/misc.go:54]: Maybe it's just me, I'm not familiar with that abbreviation.

**Comment by knisbet** [comment]
> @knisbet @wadells PTAL - is there anything else you'd like to change?

I didn't recheck everything, but I'm good to merge.

**Review by knisbet — APPROVED** [review]

**Pull request merged**